cluster-monitoring/templates/prometheusrule.yaml

```diff
@@ -29,3 +30,36 @@ spec:
       annotations:
         description: '{{`{{$labels.instance}}`}} with calico-node pod {{`{{$labels.pod}}`}} has been having dataplane failures'
         summary: 'Instance {{`{{$labels.instance}}`}} - Dataplane failures'
+---
+apiVersion: monitoring.coreos.com/v1
+kind: PrometheusRule
+metadata:
+  name: "{{ .Release.Name }}-node-custom"
+  namespace: "{{ .Release.Namespace }}"
+  labels:
+    app: prometheus
+    prometheus: k8s-monitor
+    chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
+    release: "{{ .Release.Name }}"
+    heritage: "{{ .Release.Service }}"
+    role: alert-rules
+spec:
+  groups:
+  - name: node-custom.rules
+    rules:
+    - alert: MemoryUsageHigh
+      expr: ((node_memory_MemTotal - node_memory_MemFree - node_memory_Buffers - node_memory_Cached) / node_memory_MemTotal) * 100 >= 90
```

Comment:
I think you should sum by instance. For example if one of the node-exporter pods is killed, for a period of time you'll end up with two entries for the same instance, because the pod name is used as a label in those metrics.

@SamClinckspoor I'm not 100% certain about this, so correct me if I'm wrong 😛

Reply:
Actually, I've just tried the expression in Qover's Prometheus and that's already the case. Apparently there are two jobs defined that export those metrics, so there are two entries for each instance. That means that alerts will be triggered twice I think.

Reply:
I can set it like this `((node_memory_MemTotal{job="node-exporter"} - node_memory_MemFree{job="node-exporter"} - node_memory_Buffers{job="node-exporter"} - node_memory_Cached{job="node-exporter"}) / node_memory_MemTotal{job="node-exporter"}) * 100` but I'd rather avoid collecting twice the same metrics, no?

Reply:
I don't think it will be an issue. there is at most 1 exporter per node. if it restarts you get a new timeseries but the old one is stopped. since timeframe is 1h it won't alert right away on the new one, but keep the old alert alive until it is older than 1h.